benchmarks/benchmarks.py

```diff
@@ -75,3 +75,44 @@ def time_get_parts_python(self):
             parts.append(pygeos.get_geometry(self.multipolygons[i], range(num_parts)))
 
         parts = np.concatenate(parts)
+
+
+class STRtree:
+    """Benchmarks queries against STRtree"""
+
+    def setup(self):
+        self.polygons = pygeos.polygons(np.random.random((100, 3, 2)))
```

Comment:
Although this is an easy test creation (I also use it myself often), one concern is that this might not be typical data *for spatial indexing*, because many of those polygons intersect: from a quick test, often more than 60% of all polygons combinations in the array are overlapping (for other benchmarks, like element-wise operations where dependence between elements doesn't matter, this should not be a concern, but for a spatial index it might matter)

Now, it's probably fine to add those benchmarks anyway, but it might be good to also think about other potential (random or not) datasets.

Reply:
I was thinking, if we could add a random offset (an affine translate) to spread them out over a bit larger area, that would reduce the overlap quite a bit. Something like:

```
a, b = -2, 2
offset = (b - a) * np.random.random_sample(100) + a
pygeos.polygons(np.random.random((100, 3, 2)) + offset[:, None, None])
``